### ISSUE TITLE
Create API endpoint for [id].ts

### ISSUE THREAD

**Issue by kleneway** [PioneerSquareLabs/otto-playground#10]
Create the API endpoint documents/[id].ts that will API route for getting a specific document by ID. Be sure to add the proper extension when creating the file. Here are the instructions:

## Description
In this issue, we will create a new Next.js API endpoint `[id].ts` that will handle the GET request to fetch a specific document by its ID. This endpoint will be used by the front-end to display the details of a document when a user clicks on it from the project documents page.

## Requirements
1. Create a new file `src/pages/api/documents/[id].ts`.
2. Import the necessary dependencies:
   - `import { NextApiRequest, NextApiResponse } from 'next';`
   - `import { Document, DocumentSchema } from '../../../types';`
   - `import { supabase } from '../../../lib/supabaseClient';`
3. Define a new async function `handler` with the following signature:
   - `async function handler(req: NextApiRequest, res: NextApiResponse): Promise<void> { ... }`
4. Inside the `handler` function, check if the request method is `GET`. If not, return a 405 status (Method Not Allowed) with a JSON object containing an error message:
   - `{ error: 'Method not allowed' }`
5. Extract the `id` parameter from the request query object (`req.query.id`) and parse it as an integer. If it's not a valid integer, return a 400 status (Bad Request) with a JSON object containing an error message:
   - `{ error: 'Invalid document ID' }`
6. Use the Supabase client to query the `documents` table and fetch the document with the given `id`. If the document is not found, return a 404 status (Not Found) with a JSON object containing an error message:
   - `{ error: 'Document not found' }`
7. Validate the fetched document using the `DocumentSchema` from the `types.ts` file. If the validation fails, return a 500 status (Internal Server Error) with a JSON object containing an error message:
   - `{ error: 'Invalid document data' }`
8. If everything is successful, return a 200 status (OK) with the JSON object containing the document data.
9. Export the `handler` function as the default export of the module.

## Example TypeScript Type for Output
The expected output of the API endpoint will be a JSON object containing the document data. The TypeScript type for the output will be the `Document` type imported from the `types.ts` file.

## Error Cases
1. If the request method is not `GET`, return a 405 status (Method Not Allowed) with a JSON object containing an error message: `{ error: 'Method not allowed' }`.
2. If the `id` parameter is not a valid integer, return a 400 status (Bad Request) with a JSON object containing an error message: `{ error: 'Invalid document ID' }`.
3. If the document is not found, return a 404 status (Not Found) with a JSON object containing an error message: `{ error: 'Document not found' }`.
4. If the fetched document fails validation, return a 500 status (Internal Server Error) with a JSON object containing an error message: `{ error: 'Invalid document data' }`.

## Summary
To summarize, we need to create a new Next.js API endpoint `[id].ts` that will handle the GET request to fetch a specific document by its ID. The endpoint should validate the input, fetch the document from the database using Supabase, validate the fetched document using Zod, and return the document data as a JSON object with the appropriate status code and error messages.
